pkg/network/bqueue/queue.go

```diff
@@ -125,20 +152,40 @@ func (bq *Queue) Run() {
 	}
 }
 
-// PutBlock enqueues block to be added to the chain.
-func (bq *Queue) PutBlock(block *block.Block) error {
-	h := bq.chain.BlockHeight()
+func (bl *Queue) add(item Queueble) error {
+	if b, ok := item.(*block.Block); ok {
+		return bl.chain.AddBlock(b)
+	}
+	if h, ok := item.(*block.Header); ok {
+		return bl.chain.AddHeaders(h)
+	}
+	return nil
+}
```

Comment:
NACK. I literally mean make the queue generic:
```
var bQueue := New[Block](blockAdapter{chain}, ...)
var bSyncQueue := New[Block](blockAdapter{stateSyncModule}, ...)
var hSyncQueue := New[Header](headerAdapter{stateSyncModule}, ...)
```
With `func New[Q Queueable](bc Blockqueuer[Q], ...)`. Then implement `Blockqueuer` over `blockAdapter` and `headerAdapter`. And then inside the queue use `bq.chain.AddItem(item)` for a single block/header  or `bq.chain.AddItems(item)` for a multiple headers, that's it.

You will need to modify Blockqueuer:
```
type Blockqueuer[Q Queueable] interface {
	AddItem(Q) error
	AddItems(...Q) error
	Height() uint32
}
```